src/uwtools/rocoto.py

```diff
@@ -184,38 +193,53 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param e: The parent element to add the new element to.
         :param config: Configuration data for this element.
         """
-        operands, operators, strequality = self._dependency_constants
+        operators, strequality = self._dependency_constants
         e = SubElement(e, STR.dependency)
         for tag, subconfig in config.items():
             tag, _ = self._tag_name(tag)
-            if tag in operands:
-                self._add_task_dependency_operand_operator(e, config={tag: subconfig})
-            elif tag in operators:
-                self._add_task_dependency_operand_operator(e, config={tag: subconfig})
+            if tag in operators:
+                self._add_task_dependency_operator(e, config={tag: subconfig})
             elif tag in strequality:
                 self._add_task_dependency_strequality(e, subconfig=subconfig, tag=tag)
+            elif tag == STR.datadep:
+                self._add_task_dependency_datadep(e, subconfig)
+            elif tag == STR.taskdep:
+                self._add_task_dependency_taskdep(e, subconfig)
+            elif tag == STR.timedep:
+                self._add_task_dependency_timedep(e, subconfig)
             else:
                 raise UWConfigError("Unhandled dependency type %s" % tag)
 
-    def _add_task_dependency_operand_operator(self, e: Element, config: dict) -> None:
+    def _add_task_dependency_datadep(self, e: Element, config: dict) -> None:
         """
-        Add an operand or operator element to the <dependency>.
+        Add a <datadep> element to the <dependency>.
 
         :param e: The parent element to add the new element to.
         :param config: Configuration data for this element.
         """
-        operands, operators, strequality = self._dependency_constants
+        e = self._add_compound_time_string(e, config[STR.value], STR.datadep)
+        self._set_attrs(e, config)
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
```

Comment:
This method now looks suspiciously similar to `_add_task_dependency()` -- I'll see if I can DRY them out.

Reply:
These are DRYed out now.